MoltenVK/MoltenVK/GPUObjects/MVKSwapchain.mm

```diff
@@ -190,8 +190,8 @@
 VkResult MVKSwapchain::getRefreshCycleDuration(VkRefreshCycleDurationGOOGLE *pRefreshCycleDuration) {
 	if (_device->getConfigurationResult() != VK_SUCCESS) { return _device->getConfigurationResult(); }
 
-	auto* screen = getCAMetalLayer().screenMVK;		// Will be nil if headless
 #if MVK_MACOS && !MVK_MACCAT
+    auto* screen = getCAMetalLayer().screenMVK;        // Will be nil if headless
```

Comment:
Although the declaration might look a bit odd, it would probably be more consistent to add something like the following to `CAMetalLayer+MoltenVK.h/mm`:

`@property(nonatomic, readonly) nil screenMVK;`

Reply:
> ```objc
> @property(nonatomic, readonly) nil screenMVK;
> ```

That syntax doesn't work:

> ```
> test41.m:5:32: error: type name requires a specifier or qualifier
> @property(nonatomic, readonly) nil screenMVK;
>                                ^
> /usr/include/objc/./objc.h:108:16: note: expanded from macro 'nil'
> #   define nil __DARWIN_NULL
>                ^
> /usr/include/sys/_types.h:52:23: note: expanded from macro '__DARWIN_NULL'
> #define __DARWIN_NULL ((void *)0)
>                       ^
> test41.m:5:32: error: expected member name or ';' after declaration specifiers
> /usr/include/objc/./objc.h:108:16: note: expanded from macro 'nil'
> #   define nil __DARWIN_NULL
>                ^
> /usr/include/sys/_types.h:52:25: note: expanded from macro '__DARWIN_NULL'
> #define __DARWIN_NULL ((void *)0)
>                         ^
> test41.m:5:32: error: expected ')'
> /usr/include/objc/./objc.h:108:16: note: expanded from macro 'nil'
> #   define nil __DARWIN_NULL
>                ^
> /usr/include/sys/_types.h:52:25: note: expanded from macro '__DARWIN_NULL'
> #define __DARWIN_NULL ((void *)0)
>                         ^
> test41.m:5:32: note: to match this '('
> /usr/include/objc/./objc.h:108:16: note: expanded from macro 'nil'
> #   define nil __DARWIN_NULL
>                ^
> /usr/include/sys/_types.h:52:24: note: expanded from macro '__DARWIN_NULL'
> #define __DARWIN_NULL ((void *)0)
>                        ^
> test41.m:5:32: error: expected ')'
> @property(nonatomic, readonly) nil bar;
>                                ^
> /usr/include/objc/./objc.h:108:16: note: expanded from macro 'nil'
> #   define nil __DARWIN_NULL
>                ^
> /usr/include/sys/_types.h:52:32: note: expanded from macro '__DARWIN_NULL'
> #define __DARWIN_NULL ((void *)0)
>                                ^
> test41.m:5:32: note: to match this '('
> /usr/include/objc/./objc.h:108:16: note: expanded from macro 'nil'
> #   define nil __DARWIN_NULL
>                ^
> /usr/include/sys/_types.h:52:23: note: expanded from macro '__DARWIN_NULL'
> #define __DARWIN_NULL ((void *)0)
>                       ^
> test41.m:5:1: error: property requires fields to be named
> @property(nonatomic, readonly) nil screenMVK;
> ^                              ~~~
> test41.m:5:35: error: expected ';' at end of declaration list
> @property(nonatomic, readonly) nil screenMVK;
>                                   ^
>                                   ;
> test41.m:5:36: warning: type specifier missing, defaults to 'int'
>       [-Wimplicit-int]
> @property(nonatomic, readonly) nil screenMVK;
>                                    ^
> test41.m:5:36: error: cannot declare variable inside @interface or @protocol
> 1 warning and 7 errors generated.
> ```

It doesn't even work with Xcode 15.3β3.

Reply:
Thanks for testing. 

I haven't checked, but perhaps there is an equivalent object type in VisionOS that could be declared to be returned, even if we do return `nil`? 

If there isn't something equivalent that makes sense, we can ignore my suggestion here.